File: src/Mine/fragile_mine.py
from mine import Mine

class FragileMine(Mine):
    """ Represents a Fragile Mine """
    fragile = True
    
    def defuse(self, drone):
        """ Defuse the Mine """
        Mine.scan(self, drone)

    def fragileDefuse(self, drone):
        """ Defuse a Fragile Mine """
        Mine.defuse(self, drone)

File: src/android/src/com/kaottessur/pymine/defense/FragileMine.java
package com.kaottessur.pymine.defense;

import com.kaottessur.pymine.Power;
import com.kaottessur.pymine.level.Drone;

public class FragileMine extends Mine {
	@Override
	public boolean isFragile() {
		return true;
	}
	
	@Override
	public int getPowerRating() {
		return Power.CAREFUL_DEFUSE_POWER;
	}
	
	@Override
	public void defuse(Drone drone) {
		tryToDetonate(drone);
	}
}
